cnf-certification-test/platform/suite_test.go

```diff
@@ -15,3 +15,226 @@
+			taintInfoFuncRet: "4096",
+			taintInfoFuncErr: nil,
+
+			modulesFromNodeFuncRet: []string{"vboxsf"},
+			outOfTreeFuncRet:       []string{"vboxsf"},
+			modulesInTreeFuncRet:   false,
+			acceptedModule:         "vboxsf", // acceptable
+
+			expectedGetKernelTaintInfoCalls: 1,
+			expectedGetModulesFromNodeCalls: 1,
+			expectedModuleInTreeCalls:       1,
+			expectedGetOutOfTreeModules:     1,
+
+			expectedResult: true, // Pass because accepted module
+		},
+		{ // Test Case #4 - Tainted node, not accepted
+			taintInfoFuncRet: "4096",
+			taintInfoFuncErr: nil,
+
+			modulesFromNodeFuncRet: []string{"vboxsf"},
+			outOfTreeFuncRet:       []string{"vboxsf"},
+			modulesInTreeFuncRet:   false,
+			acceptedModule:         "", // no modules accepted
+
+			expectedGetKernelTaintInfoCalls: 1,
+			expectedGetModulesFromNodeCalls: 1,
+			expectedModuleInTreeCalls:       1,
+			expectedGetOutOfTreeModules:     1,
+
+			expectedResult: false, // Fail because not-accepted
+		},
+		{ // Test Case #5 - Tainted node with multiple taints different reasons including `module was loaded`
+			// Kernel is Tainted for following reasons:
+			// * Proprietary module was loaded (#0)
+			// * Kernel issued warning (#9)
+			// * Externally-built ('out-of-tree') module was loaded  (#12)
+			taintInfoFuncRet: "4609",
+			taintInfoFuncErr: nil,
+
+			modulesFromNodeFuncRet: []string{"vboxsf"},
+			outOfTreeFuncRet:       []string{"vboxsf"},
+			modulesInTreeFuncRet:   false,
+			acceptedModule:         "vboxsf", // no modules accepted
+
+			expectedGetKernelTaintInfoCalls: 1,
+			expectedGetModulesFromNodeCalls: 1,
+			expectedModuleInTreeCalls:       1,
+			expectedGetOutOfTreeModules:     1,
+
+			expectedResult: false, // Fail because module is accepted but there are other reasons for taints
+		},
+		{ // Test Case #6 - Failure to gather taint info from node
+			taintInfoFuncRet: "",
+			taintInfoFuncErr: errors.New("this is an error"),
+
+			modulesFromNodeFuncRet: []string{},
+			outOfTreeFuncRet:       []string{},
+			modulesInTreeFuncRet:   false, // unused
+			acceptedModule:         "",
+
+			expectedGetKernelTaintInfoCalls: 1,
+			expectedGetModulesFromNodeCalls: 0,
+			expectedModuleInTreeCalls:       0,
+			expectedGetOutOfTreeModules:     0,
+
+			expectedResult: false, // Fail because failed to query node's taint info
+		},
+		{ // Test Case #7 - Un-tainted node
+			taintInfoFuncRet: "0",
+			taintInfoFuncErr: nil,
+
+			modulesFromNodeFuncRet: []string{},
+			outOfTreeFuncRet:       []string{},
+			modulesInTreeFuncRet:   false, // unused
+			acceptedModule:         "",
+
+			expectedGetKernelTaintInfoCalls: 1,
+			expectedGetModulesFromNodeCalls: 0,
+			expectedModuleInTreeCalls:       0,
+			expectedGetOutOfTreeModules:     0,
+
+			expectedResult: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		sharedResult := true
+
+		generateEnv := func(acceptedModule string) *provider.TestEnvironment {
```

Comment:
I guess this function definition can be hoisted out of the loop as it does not depend on the specific test case.

Reply:
Well, the only reason I moved it here was to redefine/reset `sharedResult`.  Since `sharedResult` gets reset to `true` every loop I wanted to be able to set it to false again if needed.